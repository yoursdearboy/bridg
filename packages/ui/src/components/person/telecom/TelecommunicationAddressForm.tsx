import {
  Alert,
  Button,
  Group,
  LoadingOverlay,
  Select,
  Stack,
  TextInput,
} from "@mantine/core";
import { useForm } from "@mantine/form";
import type { UseMutationResult } from "@tanstack/react-query";
import {
  TelecommunicationAddressUse,
  URLScheme,
  type TelecommunicationAddress,
  type TelecommunicationAddressData,
} from "api-ts";
import { useTranslation } from "react-i18next";

interface TelecommunicationAddressFormProps {
  initialValues: TelecommunicationAddressData;
  onCancel: () => void;
  mutation: UseMutationResult<
    TelecommunicationAddress,
    Error,
    TelecommunicationAddressData,
    unknown
  >;
}

export const TelecommunicationAddressForm = ({
  initialValues,
  onCancel,
  mutation,
}: TelecommunicationAddressFormProps) => {
  const { t } = useTranslation();

  const form = useForm<TelecommunicationAddressData>({
    initialValues: {
      ...initialValues,
      use: TelecommunicationAddressUse.H,
    },
    validate: {
      scheme: (value) =>
        value
          ? null
          : t("fieldRequiredMessage", {
              fieldName: t("TelecommunicationAddress.scheme"),
            }),
      address: (value) =>
        value
          ? null
          : t("fieldRequiredMessage", {
              fieldName: t("TelecommunicationAddress.address"),
            }),
    },
  });

  const handleSubmit = (data: TelecommunicationAddressData) =>
    mutation.mutate(data);
  const schemes = Object.values(URLScheme).map((value) => ({
    label: value,
    value,
  }));
  return (
    <form onSubmit={form.onSubmit(handleSubmit)}>
      <Stack gap="md" pos="relative">
        <LoadingOverlay visible={mutation.isPending} />
        {mutation.error && (
          <Alert color="red" mb="md">
            {mutation.error.message}
          </Alert>
        )}
        <Group grow>
          <Select
            label={t("TelecommunicationAddress.scheme")}
            data={schemes}
            {...form.getInputProps("scheme")}
          />

          <TextInput
            label={t("TelecommunicationAddress.address")}
            withAsterisk
            {...form.getInputProps("address")}
          />
        </Group>

        <Group justify="flex-end" mt="md">
          <Button
            variant="outline"
            onClick={onCancel}
            disabled={mutation.isPending}
          >
            {t("cancel")}
          </Button>
          <Button type="submit" loading={mutation.isPending}>
            {t("submit")}
          </Button>
        </Group>
      </Stack>
    </form>
  );
};
