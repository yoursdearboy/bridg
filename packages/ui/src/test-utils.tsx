/* eslint-disable react-refresh/only-export-components */
import { MantineProvider } from "@mantine/core";
import { DatesProvider } from "@mantine/dates";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import {
  type AnyRoute,
  createRootRoute,
  createRootRouteWithContext,
  createRoute,
  createRouter,
  RouterProvider,
} from "@tanstack/react-router";
import { act, render } from "@testing-library/react";
import "./i18n";
import type { PropsWithChildren } from "react";

const queryClient = new QueryClient();

const App = ({ children }: React.PropsWithChildren) => {
  return (
    <QueryClientProvider client={queryClient}>
      <MantineProvider defaultColorScheme="auto">
        <DatesProvider settings={{ locale: "en" }}>{children}</DatesProvider>
      </MantineProvider>
    </QueryClientProvider>
  );
};

export const renderRoute = (
  r: AnyRoute,
  { params, props }: { params?: object; props?: PropsWithChildren } = {}
) => {
  const rootRoute = createRootRouteWithContext()();
  let routeConfig = {
    ...r.options,
    params: params,
    getParentRoute: () => rootRoute,
    path: "/",
  };
  if (props?.children)
    routeConfig = { ...routeConfig, component: () => props.children };

  const testRoute = createRoute(routeConfig);
  const routeTree = rootRoute.addChildren([testRoute]);
  const router = createRouter({
    routeTree,
    context: {
      queryClient,
    },
  });
  return act(() =>
    render(
      <App>
        <RouterProvider router={router} />
      </App>
    )
  );
};

export const renderComponent = (component: React.ReactNode) =>
  render(<App>{component}</App>);

export const EmptyRouterProvider = (props: PropsWithChildren) => {
  const rootRoute = createRootRoute({
    component: () => props.children,
  });

  return renderRoute(rootRoute, { props: { children: props.children } });
};
