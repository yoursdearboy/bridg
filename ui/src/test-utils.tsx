import { MantineProvider } from "@mantine/core";
import { DatesProvider } from "@mantine/dates";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import {
  createRootRoute,
  createRoute,
  createRouter,
  RouterProvider,
  type AnyRoute,
} from "@tanstack/react-router";
import { act, render } from "@testing-library/react";
import "./i18n";

// eslint-disable-next-line react-refresh/only-export-components
const App = ({ children }: React.PropsWithChildren) => {
  const queryClient = new QueryClient();

  return (
    <QueryClientProvider client={queryClient}>
      <MantineProvider defaultColorScheme="auto">
        <DatesProvider settings={{ locale: "en" }}>{children}</DatesProvider>
      </MantineProvider>
    </QueryClientProvider>
  );
};

export const renderRoute = (
  r: AnyRoute,
  { params }: { params?: object } = {}
) => {
  const rootRoute = createRootRoute();
  const testRoute = createRoute({
    ...r.options,
    params: params,
    getParentRoute: () => rootRoute,
    path: "/",
  });
  const routeTree = rootRoute.addChildren([testRoute]);
  const router = createRouter({
    routeTree,
  });
  return act(() =>
    render(
      <App>
        <RouterProvider router={router} />
      </App>
    )
  );
};
