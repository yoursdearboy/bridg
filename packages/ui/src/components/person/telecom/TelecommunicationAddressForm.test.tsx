import { type UseMutationResult } from "@tanstack/react-query";
import {
  type TelecommunicationAddress,
  type TelecommunicationAddressData,
} from "api-ts";
import { describe, expect, it, vi } from "vitest";
import { renderComponent } from "@/test-utils";
import { TelecommunicationAddressForm } from "./TelecommunicationAddressForm";

describe("TelecommunicationAddressForm", () => {
  it("matches snapshot", () => {
    const initialValues: TelecommunicationAddressData = {
      address: "Pizza",
      scheme: "ftp",
      use: "H",
    };
    const result = {
      id: "1",
      label: "ftp Pizza",
      ...initialValues,
    };
    const mockMutation: UseMutationResult<
      TelecommunicationAddress,
      Error,
      TelecommunicationAddressData,
      unknown
    > = {
      data: result,
      error: null,
      isError: false,
      isIdle: false,
      isPending: false,
      isPaused: false,
      isSuccess: true,
      failureCount: 0,
      failureReason: null,
      mutate: vi.fn(),
      mutateAsync: vi.fn(),
      reset: vi.fn(),
      submittedAt: 0,
      variables: {},
      status: "success",
      context: {},
    };

    const { asFragment } = renderComponent(
      <TelecommunicationAddressForm
        mutation={mockMutation}
        initialValues={initialValues}
        onCancel={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
