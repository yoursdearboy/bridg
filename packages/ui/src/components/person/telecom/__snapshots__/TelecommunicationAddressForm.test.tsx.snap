// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TelecommunicationAddressForm > matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <form>
    <div
      class="m_6d731127 mantine-Stack-root"
      style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start; position: relative;"
    >
      <div
        class="m_6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
      >
        <div
          class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
        >
          <label
            class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
            data-required="true"
            for="mantine-4miilw5eg"
            id="mantine-4miilw5eg-label"
          >
            Address
            <span
              aria-hidden="true"
              class="m_78a94662 mantine-InputWrapper-required mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
            data-variant="default"
          >
            <input
              aria-invalid="false"
              class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
              data-path="address"
              data-variant="default"
              id="mantine-4miilw5eg"
              value="Pizza"
            />
          </div>
        </div>
        <div
          class="m_46b77525 mantine-InputWrapper-root mantine-Select-root"
        >
          <label
            class="m_8fdc1311 mantine-InputWrapper-label mantine-Select-label"
            for="mantine-pfqyzivqd"
            id="mantine-pfqyzivqd-label"
          >
            Scheme
          </label>
          <div
            class="m_6c018570 mantine-Input-wrapper mantine-Select-wrapper"
            data-pointer="true"
            data-variant="default"
            data-with-right-section="true"
            style="--input-right-section-pointer-events: none;"
          >
            <input
              aria-haspopup="listbox"
              aria-invalid="false"
              autocomplete="off"
              class="m_8fb7ebe7 mantine-Input-input mantine-Select-input"
              data-path="scheme"
              data-variant="default"
              id="mantine-pfqyzivqd"
              readonly=""
              value="FTP"
            />
            <div
              class="m_82577fc2 mantine-Input-section mantine-Select-section"
              data-position="right"
            >
              <svg
                class="m_2943220b mantine-ComboboxChevron-chevron"
                data-combobox-chevron="true"
                fill="none"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
        <input
          type="hidden"
          value="ftp"
        />
      </div>
      <div
        class="m_4081bf90 mantine-Group-root"
        style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap; margin-top: var(--mantine-spacing-md);"
      >
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-variant="outline"
          style="--button-bg: transparent; --button-hover: var(--mantine-color-blue-outline-hover); --button-color: var(--mantine-color-blue-outline); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-blue-outline);"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              Cancel
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          style="--button-color: var(--mantine-color-white);"
          type="submit"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              Submit
            </span>
          </span>
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
